src/pack.rs

```diff
@@ -328,6 +343,130 @@ async fn archive_directory(input_dir: &Path, archive_target: &Path) -> Result<()
+        windows_header
+    } else {
+        unix_header
+    };
+
+    let executable_path = target.with_extension(if platform.is_windows() { "ps1" } else { "sh" });
+
+    // Determine the target OS and architecture
+    let (os, arch) = match platform {
+        Platform::Linux64 => ("unknown-linux-musl", "x86_64"),
+        Platform::LinuxAarch64 => ("unknown-linux-musl", "aarch64"),
+        Platform::Osx64 => ("apple-darwin", "x86_64"),
+        Platform::OsxArm64 => ("apple-darwin", "aarch64"),
+        Platform::Win64 => ("pc-windows-msvc", "x86_64"),
+        Platform::WinArm64 => ("pc-windows-msvc", "aarch64"),
+        _ => return Err(anyhow!("Unsupported platform: {}", platform)),
+    };
+
+    let executable_name = format!("pixi-pack-{}-{}", arch, os);
+    let extension = if platform.is_windows() { ".exe" } else { "" };
+
+    let version = env!("CARGO_PKG_VERSION");
+    let url = format!(
+        "https://github.com/Quantco/pixi-pack/releases/download/v{}/{}{}",
+        version, executable_name, extension
+    );
+
+    eprintln!("📥 Downloading pixi-pack executable...");
+    let client = reqwest::Client::new();
+    let response = client.get(&url).send().await?;
+    if !response.status().is_success() {
+        return Err(anyhow!(
+            "Failed to download pixi-pack executable. Status: {}",
+            response.status()
+        ));
+    }
+
+    let total_size = response
+        .content_length()
+        .ok_or_else(|| anyhow!("Failed to get content length"))?;
+
+    let bar = ProgressReporter::new(total_size);
+    bar.pb.set_message("Downloading");
+
+    let mut executable_bytes = Vec::new();
+    let mut stream = response.bytes_stream();
+
+    while let Some(chunk) = stream.next().await {
+        let chunk = chunk?;
+        executable_bytes.extend_from_slice(&chunk);
+        bar.pb.inc(chunk.len() as u64);
+    }
+
+    bar.pb.finish_with_message("Download complete");
+
+    eprintln!("✅ Pixi-pack executable downloaded successfully");
+
+    let mut final_executable = File::create(&executable_path)
+        .await
+        .map_err(|e| anyhow!("could not create final executable file: {}", e))?;
+
+    final_executable.write_all(header.as_bytes()).await?;
+    final_executable.write_all(b"\n").await?; // Add a newline after the header
+
+    // Encode the archive to base64
+    let archive_base64 = STANDARD.encode(&compressor);
+    final_executable
+        .write_all(archive_base64.as_bytes())
+        .await?;
+
+    final_executable.write_all(b"\n").await?;
+    if platform.is_windows() {
+        final_executable.write_all(b"__END_ARCHIVE__\n").await?;
+    } else {
+        final_executable.write_all(b"@@END_ARCHIVE@@\n").await?;
+    }
+
+    // Encode the executable to base64
+    let executable_base64 = STANDARD.encode(&executable_bytes);
+    final_executable
+        .write_all(executable_base64.as_bytes())
+        .await?;
+
+    // Make the script executable
```

Comment:
let's add a comment that this won't be executed when cross-packing but is not possible otherwise since the FS most likely doesn't support it